dev/ProgressRing/ProgressRing.h

```diff
@@ -21,25 +23,36 @@ class ProgressRing :
     void OnApplyTemplate();
 
     void OnStrokeThicknessPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&);
+    void OnBackgroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&);
 
 private:
     void OnRangeBasePropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&);
-
     void OnSizeChanged(const winrt::IInspectable&, const winrt::IInspectable&);
+    void ApplyLottieAnimation();
+    void UpdateStates();
     void UpdateSegment();
     void UpdateRing();
 
-    static winrt::Size ComputeEllipseSize(double thickness, double width, double height);
+    static winrt::Size ComputeCircleSize(double thickness, double width);
```

Comment:
>static [](start = 4, length = 6)

Remove from the header. It doesn't care about any state of the ProgressRing class, so it's just a module-specific function - no need to make it a member.
Likewise for the static constexprs below. They all belong in the .cpp.